tests/data/test_publish.py

```diff
@@ -296,3 +311,37 @@ def test_transform_should_filter_ids(self, caplog: pytest.LogCaptureFixture):
             is_renting=True,
         )
         pd.testing.assert_frame_equal(actual.reset_index(drop=True), expected)
+
+
+class TestFileLoader:
+    def test_load(self, mocker: MockerFixture):
+        mock_to_pickle = mocker.patch.object(pd.DataFrame, "to_pickle")
+        mocker.patch.object(Path, "is_file", return_value=True)
+        fake_output_path = Path("fake_filepath")
+        loader = FileLoader(output_file=fake_output_path)
+
+        loader.load(pd.DataFrame())
+        mock_to_pickle.assert_called_once_with(fake_output_path)
+
+
+class TestSQLLoader(TestDatabaseInteraction):
+    def test_load(self, mocker: MockerFixture):
+        data_to_load = pd.DataFrame.from_records(
+            [
+                [1, datetime(2023, 12, 10, 12, 0), 5],
+            ],
+            columns=["station_id", "datetime", "num_bikes_available"],
+        ).assign(
+            num_bikes_available_types_mechanical=2,
+            num_bikes_available_types_ebike=3,
+            num_docks_available=5,
+            is_installed=True,
+            is_returning=True,
+            is_renting=True,
+        )
+        mock_to_sql = mocker.patch.object(pd.DataFrame, "to_sql")
+        fake_table_name = ""
+        loader = SQLLoader(table_name=fake_table_name)
+        loader.load(data_to_load)
+
+        mock_to_sql.assert_called_once()
```

Comment:
Well done!
Maybe you could specify what is passed to the `to_sql` function with `assert_called_once_with`